standardtracer/propagation.go

```diff
@@ -44,9 +45,11 @@ func (p *splitTextPropagator) InjectSpan(
 		fieldNameSampled: strconv.FormatBool(sc.raw.Sampled),
```

Comment:
on line 37 (github won't let me comment there):
```go
sc := sp.(*spanImpl)
```
Maybe there is another type of std error the injectors could return, i.e.
```go
sc, ok := sp.(*spanImpl)
if !ok {
  return ErrWrongSpanType
}
```